src/openrct2/ride/Ride.cpp

```diff
@@ -2163,10 +2163,11 @@ void ride_set_vehicle_colours_to_random_preset(Ride* ride, uint8_t preset_index)
     else
     {
         ride->colour_scheme_type = RIDE_COLOUR_SCHEME_DIFFERENT_PER_TRAIN;
-        uint32_t count = std::min(presetList->count, static_cast<uint8_t>(32));
+        uint32_t count = presetList->count;
         for (uint32_t i = 0; i < count; i++)
```

Comment:
This doesn't look quite right. We want to wrap when we have exhausted the preset list size not when we have exhausted 32 trains. (I think)

Reply:
I'm not sure I understand. Changing `count` to be `presetList->count` makes it so we will iterate and fill more than 32 vehicle_colour objects, which gives us the wanted result.
If you refer to the modulo computation, `presetList->count` is 255, so modulo won't work with it.
How do you suggest we improve it?

Reply:
ignore.